repos/system_upgrade/common/actors/initramfs/checkinitramfstasks/libraries/checkinitramfstasks.py

```diff
@@ -22,51 +22,65 @@ def _printable_modules(conflicts):
+
+
+def _detect_modules_conflicts(msgtype, kind):
     """
     Return dict of modules with conflicting tasks
 
-    In this case when a dracut module should be applied but different
-    sources are specified. E.g.:
-       include dracut modules X where,
+    In this case when a module should be applied but different sources are
+    specified. E.g.:
+       include modules X where,
          msg A)  X
          msg B)  X from custom path
     """
-    dracut_modules = defaultdict(set)
+
+    msgattr = None
+    if kind == 'dracut':
+        msgattr = 'include_dracut_modules'
+        treat_path_fn = _treat_path_dracut
+    elif kind == 'kernel':
+        msgattr = 'include_kernel_modules'
+        treat_path_fn = _treat_path_kernel
+
+    modules = defaultdict(set)
     for msg in api.consume(msgtype):
-        for dmodule in msg.include_dracut_modules:
-            dracut_modules[dmodule.name].add(_treat_path(dmodule))
-    return {key: val for key, val in dracut_modules.items() if len(val) > 1}
+        for module in getattr(msg, msgattr):
+            modules[module.name].add(treat_path_fn(module))
+    return {key: val for key, val in modules.items() if len(val) > 1}
 
 
 def process():
-    conflicts = _detect_dracut_modules_conflicts(UpgradeInitramfsTasks)
-    if conflicts:
-        report = [
-            reporting.Title('Conflicting requirements of dracut modules for the upgrade initramfs'),
-            reporting.Summary(SUMMARY_DRACUT_FMT.format(_printable_modules(conflicts))),
-            reporting.Severity(reporting.Severity.HIGH),
-            reporting.Groups([reporting.Groups.SANITY]),
-            reporting.Groups([reporting.Groups.INHIBITOR]),
-        ]
-        reporting.create_report(report)
-
-    conflicts = _detect_dracut_modules_conflicts(TargetInitramfsTasks)
-    if conflicts:
-        report = [
-            reporting.Title('Conflicting requirements of dracut modules for the target initramfs'),
-            reporting.Summary(SUMMARY_DRACUT_FMT.format(_printable_modules(conflicts))),
-            reporting.Severity(reporting.Severity.HIGH),
-            reporting.Groups([reporting.Groups.SANITY]),
-            reporting.Groups([reporting.Groups.INHIBITOR]),
-        ]
-        reporting.create_report(report)
+    groups = [('upgrade', 'kernel', UpgradeInitramfsTasks), ('upgrade', 'dracut', UpgradeInitramfsTasks),
+              ('target', 'dracut', TargetInitramfsTasks)]
+    for (msgname, kind, msgtype) in groups:
+        conflicts = _detect_modules_conflicts(msgtype, kind)
+        if conflicts:
+            report = [
+                reporting.Title('Conflicting requirements of {kind} modules for the {msgname} initramfs'.format(
+                    kind=kind, msgname=msgname)),
+                reporting.Summary(SUMMARY_FMT.format(kind=kind, conflicts=_printable_modules(conflicts))),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Groups([reporting.Groups.SANITY]),
```

Comment:
Listing all groups on one line should be ok
`reporting.Groups([reporting.Groups.SANITY, reporting.Groups.INHIBITOR])`

Reply:
Right. Fixed!